src/bindgen/ir/constant.rs

```diff
@@ -45,7 +49,8 @@ impl Literal {
             Literal::PostfixUnaryOp { .. }
             | Literal::BinOp { .. }
             | Literal::Expr(..)
-            | Literal::Path(..) => {}
+            | Literal::Path(..)
+            | Literal::Cast { .. } => {}
```

Comment:
This needs to call `replace_self_with` in the subexpression, but this is also an issue with other literals.